### PR TITLE
Feature/class fields docs

### PR DESCRIPTION
**Pull request by sdepold**
copy of #13861

### CODE REVIEW

File: docs/manual/other-topics/typescript.md [hunk 1]
```diff
@@ -46,28 +44,28 @@ interface UserCreationAttributes extends Optional<UserAttributes, "id"> {}
 
 class User extends Model<UserAttributes, UserCreationAttributes>
   implements UserAttributes {
-  public id!: number; // Note that the `null assertion` `!` is required in strict mode.
-  public name!: string;
-  public preferredName!: string | null; // for nullable fields
+  declare id: number; // Note that the `null assertion` `!` is required in strict mode.
```
Comment (reduardo7) [inline docs/manual/other-topics/typescript.md:47]: Remove `// Note that the `null assertion` `!` is required in strict mode.` comment.

File: docs/manual/other-topics/typescript.md [hunk 2]
```diff
@@ -46,28 +44,28 @@ interface UserCreationAttributes extends Optional<UserAttributes, "id"> {}
...
 
   // Since TS cannot determine model association at compile time
   // we have to declare them here purely virtually
   // these will not exist until `Model.init` was called.
-  public getProjects!: HasManyGetAssociationsMixin<Project>; // Note the null assertions!
-  public addProject!: HasManyAddAssociationMixin<Project, number>;
-  public hasProject!: HasManyHasAssociationMixin<Project, number>;
-  public countProjects!: HasManyCountAssociationsMixin;
-  public createProject!: HasManyCreateAssociationMixin<Project>;
+  declare getProjects: HasManyGetAssociationsMixin<Project>; // Note the null assertions!
```
Comment (reduardo7) [inline docs/manual/other-topics/typescript.md:58]: Comment?

File: types/test/typescriptDocs/ModelInit.ts [hunk 3]
```diff
@@ -21,28 +21,28 @@ interface UserCreationAttributes extends Optional<UserAttributes, "id"> {}
 
 class User extends Model<UserAttributes, UserCreationAttributes>
   implements UserAttributes {
-  public id!: number; // Note that the `null assertion` `!` is required in strict mode.
-  public name!: string;
-  public preferredName!: string | null; // for nullable fields
+  declare id: number; // Note that the `null assertion` `!` is required in strict mode.
```
Comment (reduardo7) [inline types/test/typescriptDocs/ModelInit.ts:24]: Remove `Note that the `null assertion` `!` is required in strict mode.` comment.

File: types/test/typescriptDocs/ModelInit.ts [hunk 4]
```diff
@@ -21,28 +21,28 @@ interface UserCreationAttributes extends Optional<UserAttributes, "id"> {}
...
 
   // Since TS cannot determine model association at compile time
   // we have to declare them here purely virtually
   // these will not exist until `Model.init` was called.
-  public getProjects!: HasManyGetAssociationsMixin<Project>; // Note the null assertions!
-  public addProject!: HasManyAddAssociationMixin<Project, number>;
-  public hasProject!: HasManyHasAssociationMixin<Project, number>;
-  public countProjects!: HasManyCountAssociationsMixin;
-  public createProject!: HasManyCreateAssociationMixin<Project>;
+  declare getProjects: HasManyGetAssociationsMixin<Project>; // Note the null assertions!
```
Comment (reduardo7) [inline types/test/typescriptDocs/ModelInit.ts:35]: Comment valid?

File: docs/manual/other-topics/typescript.md (hunk 1 above)
Comment (ephys) [reply docs/manual/other-topics/typescript.md:47]: Oh yeah good catch. Do you want to open a follow-up PR?

**Pull request merged**